ubyssey/templates/objects/archive.html

```diff
@@ -0,0 +1,198 @@
+                    <a href="?{% if year %}year={{ year }}&{% endif %}{% if q %}q={{ q }}&{% endif %}section_id={{ section.id }}">{{ section.name }}</a>
+                  </li>
+                {% endfor %}
+                </ul>
+              </div>
+            </div>
+          </div>
+
+        </div>
+        {% endif %}
+
+        <div class="o-mobile-dropdown js-dropdown">
+
+          <a class="o-mobile-dropdown__button js-disable-scroll" href="#">
+            <span>{{ order|title }}</span>
+            <i class="fa fa-sort-desc"></i>
+          </a>
+
+          <div class="o-mobile-dropdown__list js-dropdown-list">
+            <div class="o-mobile-dropdown__list__container js-dropdown-container">
+              <div>
+                <div class="o-mobile-dropdown__header">
+                  <h3 class="o-mobile-dropdown__title">Order by:</h3>
+                  <div class="o-mobile-dropdown__close">
+                    <i class="fa fa-close"></i>
+                  </div>
+                </div>
+                <ul>
+                  <li class="o-mobile-dropdown__item{% if order == 'newest' %} o-mobile-dropdown__item--is-active{% endif %}">
+                    <a href="?{% if q %}q={{ q }}&{% endif %}{% if section_id %}section_id={{ section_id}}&{% endif %}{% if year %}year={{ year}}&{% endif %}order=newest">Newest</a>
+                  </li>
+                  <li class="o-mobile-dropdown__item{% if order == 'oldest' %} o-mobile-dropdown__item--is-active{% endif %}">
+                    <a href="?{% if q %}q={{ q }}&{% endif %}{% if section_id %}section_id={{ section_id}}&{% endif %}{% if year %}year={{ year}}&{% endif %}order=oldest">Oldest</a>
+                  </li>
+                </ul>
+              </div>
+            </div>
+          </div>
+
+        </div>
+
+      </div>
+
+      <div class="o-archive__header">
+        <h3 class="o-archive__header__title">
+          {% if articles %} {{ articles|length }} RESULTS {% else %}NO RESULTS{% endif %}
+        </h3>
+        <div class="o-archive__header__sort o-dropdown js-dropdown">
+          <a class="o-button o-dropdown__button">
+            <span>{{ order|title }}</span>
+            <i class="fa fa-sort-desc"></i>
+          </a>
+          <ul class="o-dropdown__list js-dropdown-list">
+            <li class="o-dropdown__item{% if order == 'newest' %} o-dropdown__item--is-active{% endif %}">
+              <a href="?{% if q %}q={{ q }}&{% endif %}{% if section_id %}section_id={{ section_id}}&{% endif %}{% if year %}year={{ year}}&{% endif %}order=newest">Newest</a>
+            </li>
+            <li class="o-dropdown__item{% if order == 'oldest' %} o-dropdown__item--is-active{% endif %}">
+              <a href="?{% if q %}q={{ q }}&{% endif %}{% if section_id %}section_id={{ section_id}}&{% endif %}{% if year %}year={{ year}}&{% endif %}order=oldest">Oldest</a>
```

Comment:
can none of these querystrings be generated by the view, at least mostly? it's hard to read what is actually changing between lines when there is so much that is so similar, and easy to make mistakes.